odoo/http.py

```diff
@@ -1366,9 +1366,16 @@ def setup_db(self, httprequest):
             httprequest.session.db = db_monodb(httprequest)
 
     def setup_lang(self, httprequest):
-        if not "lang" in httprequest.session.context:
-            lang = httprequest.accept_languages.best or "en_US"
-            lang = babel.core.LOCALE_ALIASES.get(lang, lang).replace('-', '_')
+        if "lang" not in httprequest.session.context:
+            alang = httprequest.accept_languages.best or "en-US"
+            try:
+                code, territory, _, _ = babel.core.parse_locale(alang, sep='-')
+                if territory:
+                    lang = '%s_%s' % (code, territory)
+                else:
+                    lang = babel.code.LOCALE_ALIASES[code]
```

Comment:
This line has a programming error, the module babel has no attribute code. Correct is:
`lang = babel.core.LOCALE_ALIASES[code]`